src/plots/plots.js

```diff
@@ -3181,16 +3180,19 @@ function sortAxisCategoriesByValue(axList, gd) {
                         }
                     } else {
                         // For all other 2d cartesian traces
-                        if(axLetter === 'x') {
-                            cat = cdi.p + 1 ? cdi.p : cdi.x;
-                            value = cdi.s || cdi.v || cdi.y;
```

Comment:
@antoinerg please note `cdi.s` here could be 0 and it is a valid value although it is zero.
We should have not used `cdi.v` or `cdi.(x|y)` instead in that case.

Reply:
Nice catch!